src/DTMFSender.ts

```diff
@@ -0,0 +1,113 @@
+class DTMFSender {
+  private _outputNode: MediaStreamAudioDestinationNode;
+  private _outputStream: MediaStream;
+  private _gain: GainNode;
+  private _filter: BiquadFilterNode;
+  private _sourceNode: MediaStreamAudioSourceNode;
+  private _osc1: OscillatorNode;
+  private _osc2: OscillatorNode;
+  private _toneDuration: number = 1500;
+  private _toneGap: number = 70;
+  private _tone: string = "";
+  private _playing: boolean = false;
+
+  private _dtmfFreq: Map<string, Array<number>> = new Map([
+    ["1", [1209, 697]],
+    ["2", [1336, 697]],
+    ["3", [1477, 697]],
+    ["4", [1209, 770]],
+    ["5", [1336, 770]],
+    ["6", [1477, 770]],
+    ["7", [1209, 852]],
+    ["8", [1336, 852]],
+    ["9", [1477, 852]],
+    ["*", [1209, 941]],
+    ["0", [1336, 941]],
+    ["#", [1477, 941]]
+  ]);
+
+  constructor(sender: RTCRtpSender) {
+    var audioCtx = new AudioContext();
```

Comment:
In general, I'd recommend using `let` or `const` anywhere I can as opposed to `var` to avoid accidental closure issues or redeclaring something that wasn't supposed to change.